gapic/src/main/com/google/gapid/models/ApiState.java

```diff
@@ -53,16 +54,20 @@
 
   private final ConstantSets constants;
   private final ObjectStore<Path.Any> selection = ObjectStore.create();
+  private final ApiContext contexts;
 
   public ApiState(
-      Shell shell, Client client, Follower follower, AtomStream atoms, ConstantSets constants) {
+      Shell shell, Client client, Follower follower, AtomStream atoms, ApiContext contexts, ConstantSets constants) {
     super(LOG, shell, client, Listener.class);
     this.constants = constants;
+    this.contexts = contexts;
 
     atoms.addListener(new AtomStream.Listener() {
       @Override
       public void onAtomsSelected(AtomIndex index) {
-        load(Paths.any(stateTree(index.getCommand())), false);
+        Path.StateTree path = stateTree(index.getCommand());
```

Comment:
Instead of building the proto twice, simply pass the context into the `stateTree` method (see the `commandTree` method)

Reply:
Done.